networkx/algorithms/tests/test_dag.py

```diff
@@ -11,36 +12,15 @@ def setUp(self):
         pass
 
     def test_topological_sort1(self):
-        DG = nx.DiGraph()
-        DG.add_edges_from([(1, 2), (1, 3), (2, 3)])
-        assert_equal(nx.topological_sort(DG), [1, 2, 3])
-        assert_equal(nx.topological_sort_recursive(DG), [1, 2, 3])
+        DG = nx.DiGraph([(1, 2), (1, 3), (2, 3)])
 
-        DG.add_edge(3, 2)
-        assert_raises(nx.NetworkXUnfeasible, nx.topological_sort, DG)
-        assert_raises(nx.NetworkXUnfeasible, nx.topological_sort_recursive, DG)
-
-        DG.remove_edge(2, 3)
-        assert_equal(nx.topological_sort(DG), [1, 3, 2])
-        assert_equal(nx.topological_sort_recursive(DG), [1, 3, 2])
-
-    def test_reverse_topological_sort1(self):
-        DG = nx.DiGraph()
-        DG.add_edges_from([(1, 2), (1, 3), (2, 3)])
-        assert_equal(nx.topological_sort(DG, reverse=True), [3, 2, 1])
-        assert_equal(
-            nx.topological_sort_recursive(DG, reverse=True), [3, 2, 1])
+        consume(nx.topological_sort(DG))
 
         DG.add_edge(3, 2)
-        assert_raises(nx.NetworkXUnfeasible,
-                      nx.topological_sort, DG, reverse=True)
-        assert_raises(nx.NetworkXUnfeasible,
-                      nx.topological_sort_recursive, DG, reverse=True)
+        assert_raises(nx.NetworkXUnfeasible, consume, nx.topological_sort(DG))
 
         DG.remove_edge(2, 3)
-        assert_equal(nx.topological_sort(DG, reverse=True), [2, 3, 1])
-        assert_equal(
-            nx.topological_sort_recursive(DG, reverse=True), [2, 3, 1])
+        consume(nx.topological_sort(DG))
```

Comment:
It looks like you no longer test the values returned by the sort... ```consume``` only steps through the iterator, so we are checking whether there are errors during iteration. 

Can we continue to test the values being returned by using ```list``` instead of consume? Or is there something else going on here...?